### ISSUE TITLE
[YSQL] ALTER TABLE command on a table dropping the view created on that table

### ISSUE THREAD

**Issue by priyanshi-yb** [yugabyte/yugabyte-db#15127]
Jira Link: [DB-4331](https://yugabyte.atlassian.net/browse/DB-4331)
### Description

While using the `ALTER TABLE` command on a table to add a `PRIMARY KEY` constraint dropping the view created on that table. Refer to the example below as can be seen that there is a NOTICE on the shell - `NOTICE:  drop cascades to view v1`.

Tried on YugabyteDB versions - 2.14.4.0 and 2.15.1.0

Example -
```
yugabyte=# CREATE TABLE view_table1 (
    id integer NOT NULL,
    first_name character varying(50),
    last_name character varying(50),
    email character varying(50),
    gender character varying(50),
    ip_address character varying(20)
);
CREATE TABLE
yugabyte=# CREATE VIEW v1 AS
 SELECT view_table1.first_name,
    view_table1.last_name
   FROM view_table1
  WHERE ((view_table1.gender)::text = 'Female'::text);
CREATE VIEW
yugabyte=# \d
            List of relations
 Schema |    Name     | Type  |  Owner   
--------+-------------+-------+----------
 public | v1          | view  | yugabyte
 public | view_table1 | table | yugabyte
(2 rows)

yugabyte=# ALTER TABLE  view_table1
    ADD CONSTRAINT view_table1_pkey PRIMARY KEY (id);
NOTICE:  drop cascades to view v1
ALTER TABLE
yugabyte=# \d
            List of relations
 Schema |    Name     | Type  |  Owner   
--------+-------------+-------+----------
 public | view_table1 | table | yugabyte
(1 row)

yugabyte=#
```

**Comment by knicely87** [comment]
Also, it seems that table GRANTS are also lost when a user drops a PK from a table.

Example:

```
yugabyte=# CREATE TABLE test(c1 INT PRIMARY KEY, c2 VARCHAR);
CREATE TABLE

yugabyte=# CREATE USER jim;
CREATE ROLE

yugabyte=# GRANT SELECT ON test TO jim;
GRANT

yugabyte=#  SELECT grantee, privilege_type, table_name FROM information_schema.role_table_grants WHERE grantee = 'jim';
 grantee | privilege_type | table_name
---------+----------------+------------
 jim     | SELECT         | test
(1 row)

yugabyte=# \d test
                     Table "public.test"
 Column |       Type        | Collation | Nullable | Default
--------+-------------------+-----------+----------+---------
 c1     | integer           |           | not null |
 c2     | character varying |           |          |
Indexes:
    "test_pkey" PRIMARY KEY, lsm (c1 HASH)

yugabyte=# SELECT 'test'::regclass::oid; -- Current Table OID
  oid
-------
 16438
(1 row)

yugabyte=# ALTER TABLE test DROP CONSTRAINT test_pkey;
NOTICE:  drop cascades to view test_vw
ALTER TABLE

yugabyte=# SELECT 'test'::regclass::oid; -- New Table OID
  oid
-------
 16447
(1 row)

yugabyte=# \d test
                     Table "public.test"
 Column |       Type        | Collation | Nullable | Default
--------+-------------------+-----------+----------+---------
 c1     | integer           |           | not null |
 c2     | character varying |           |          |
```

The table has a new OID, indicating that it was re-created...

And yikes, the SELECT privilege is gone!

```
yugabyte=# SELECT grantee, privilege_type, table_name FROM information_schema.role_table_grants WHERE grantee = 'jim';
 grantee | privilege_type | table_name
---------+----------------+------------
(0 rows)
```

**Comment by fizaaluthra** [comment]
Fixed by 63eaaaeadbf1ac25746e881a1fb397de4af6971d
